packages/lesmis/src/components/DataGrid/DataGrid.js

```diff
@@ -0,0 +1,148 @@
+        : true;
+    });
+  },
+});
+
+export const DataGrid = ({ data, columns }) => {
+  const defaultColumn = React.useMemo(
+    () => ({
+      Filter: DefaultColumnFilter,
+    }),
+    [],
+  );
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    prepareRow,
+    rows,
+    page,
+    canPreviousPage,
+    canNextPage,
+    nextPage,
+    previousPage,
+    setPageSize,
+    state: { pageIndex, pageSize },
+  } = useTable(
+    {
+      defaultColumn,
+      filterTypes,
+      columns,
+      data,
+      initialState: { pageSize: 20 },
+    },
+    useFilters,
+    useSortBy,
+    usePagination,
+  );
+
+  return (
+    <>
+      <StyledTable {...getTableProps()}>
+        <TableHead>
+          {headerGroups.map(headerGroup => (
+            <TableRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => (
+                <TableCell {...column.getHeaderProps(column.getSortByToggleProps())}>
+                  {column.render('Header')}
+                  <TableSortLabel
+                    active={column.isSorted}
+                    direction={column.isSortedDesc ? 'asc' : 'desc'}
+                  />
+                </TableCell>
+              ))}
+            </TableRow>
+          ))}
+          {headerGroups.map(headerGroup => (
+            <TableRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map(column => (
+                <StyledTableCell {...column.getHeaderProps()}>
+                  {column.canFilter ? column.render('Filter') : null}
+                </StyledTableCell>
+              ))}
+            </TableRow>
+          ))}
+        </TableHead>
+        <TableBody {...getTableBodyProps()}>
+          {page.length === 0 ? (
+            <NoDataRow colspan={columns.length} />
+          ) : (
+            page.map(row => {
+              prepareRow(row);
+              return (
+                <StyledTableRow {...row.getRowProps()}>
+                  {row.cells.map(cell => (
+                    <TableCell {...cell.getCellProps()}>{cell.render('Cell')}</TableCell>
+                  ))}
+                </StyledTableRow>
+              );
+            })
+          )}
+        </TableBody>
+      </StyledTable>
+      <Paginator
+        canNextPage={canNextPage}
+        canPreviousPage={canPreviousPage}
+        nextPage={nextPage}
+        pageIndex={pageIndex}
+        pageSize={pageSize}
+        previousPage={previousPage}
+        setPageSize={setPageSize}
+        totalCount={rows.length}
+      />
+    </>
+  );
+};
+
+DataGrid.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      firstName: PropTypes.string,
+      lastName: PropTypes.string,
+      email: PropTypes.string,
+      permissionGroupName: PropTypes.string,
+    }),
+  ),
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      accessor: PropTypes.string,
+      Header: PropTypes.string,
```

Comment:
Where does `canFilter` get applied? It doesn't appear in the shape of prop that can be passed through with each column, but that's where I'd expect it was controlled?

Reply:
Yeah good question. The canFilter prop can be set in the columns or it is true if the column has a Filter component set which in our case is applied through the `defaultColumn`.